src/DevBetterWeb.Infrastructure/Services/WebhookHandlerService.cs

```diff
@@ -87,29 +98,77 @@ public async Task HandleNewCustomerSubscriptionAsync(string json)
+        await HandleNewCustomerSubscriptionWithEmailAlreadyMember(subscriptionId, paymentAmount);
+      }
+      else
       {
-        var customerId = _paymentHandlerSubscription.GetCustomerId(subscriptionId);
-        var email = _paymentHandlerCustomer.GetCustomerEmail(customerId);
 
-        if (string.IsNullOrEmpty(email))
+        var status = _paymentHandlerSubscription.GetStatus(subscriptionId);
+
+        if (status == "active")
         {
-          throw new InvalidEmailException();
+          var customerId = _paymentHandlerSubscription.GetCustomerId(subscriptionId);
+          var email = _paymentHandlerCustomer.GetCustomerEmail(customerId);
+
+          if (string.IsNullOrEmpty(email))
+          {
+            throw new InvalidEmailException();
+          }
+
+          Invitation invite = await _newMemberService.CreateInvitationAsync(email, subscriptionId);
+
+          var webhookMessage = $"A new customer with email {email} has subscribed to DevBetter. They will be receiving a registration email.";
+          await _webhook.Send($"Webhook:\n{webhookMessage}");
+
+          await _newMemberService.SendRegistrationEmailAsync(invite);
+
+          await AddNewSubscriberBillingActivity(subscriptionId, email, paymentAmount);
         }
+      }
+    }
 
-        Invitation invite = await _newMemberService.CreateInvitationAsync(email, subscriptionId);
+    private async Task AddNewSubscriberBillingActivity(string subscriptionId, string email, decimal paymentAmount)
+    {
+      var subscriptionPlanName = _paymentHandlerSubscription.GetAssociatedProductName(subscriptionId);
+      var billingPeriod = _paymentHandlerSubscription.GetBillingPeriod(subscriptionId);
+      await _memberAddBillingActivityService.AddSubscriptionCreationBillingActivity(email, paymentAmount, subscriptionPlanName, billingPeriod);
+    }
 
-        var webhookMessage = $"A new customer with email {email} has subscribed to DevBetter. They will be receiving a registration email.";
-        await _webhook.Send($"Webhook:\n{webhookMessage}");
+    // Below two methods are for handling migration from Launchpass to Stripe
+    private async Task<bool> NewCustomerSubscriptionWithEmailAlreadyMember(string subscriptionId)
+    {
+      var customerId = _paymentHandlerSubscription.GetCustomerId(subscriptionId);
+      var email = _paymentHandlerCustomer.GetCustomerEmail(customerId);
 
-        await _newMemberService.SendRegistrationEmailAsync(invite);
+      var userIsMember = await _userLookupService.FindUserIsMemberByEmailAsync(email);
 
-        var subscriptionPlanName = _paymentHandlerSubscription.GetAssociatedProductName(subscriptionId);
-        var billingPeriod = _paymentHandlerSubscription.GetBillingPeriod(subscriptionId);
-        await _memberAddBillingActivityService.AddSubscriptionCreationBillingActivity(email, paymentAmount, subscriptionPlanName, billingPeriod);
+      if (userIsMember)
+      {
+        return true;
       }
+
+      return false;
```

Comment:
it's hard to see in the code here but if the current code says this:

```
if (userIsMember)
{
  return true;
}
return false;
```
That's the same as this:

```
return userIsMember;
```

Reply:
And if that's harder to read for you, what about:

```
return userIsMember == true;
```

Which is also equivalent.